libraries/libfc/include/fc/time.hpp

```diff
@@ -105,15 +104,14 @@ namespace fc {
         constexpr time_point_sec   operator +( uint32_t offset )const { return time_point_sec(utc_seconds + offset); }
         constexpr time_point_sec   operator -( uint32_t offset )const { return time_point_sec(utc_seconds - offset); }
 
-        friend constexpr time_point   operator + ( const time_point_sec& t, const microseconds& m )   { return time_point(t) + m;             }
-        friend constexpr time_point   operator - ( const time_point_sec& t, const microseconds& m )   { return time_point(t) - m;             }
-        friend constexpr microseconds operator - ( const time_point_sec& t, const time_point_sec& m ) { return time_point(t) - time_point(m); }
-        friend constexpr microseconds operator - ( const time_point& t, const time_point_sec& m ) { return time_point(t) - time_point(m); }
+        friend constexpr time_point   operator + ( const time_point_sec& t, const microseconds& m )   { return time_point{fc::seconds(t.utc_seconds) + m}; }
+        friend constexpr time_point   operator - ( const time_point_sec& t, const microseconds& m )   { return time_point{fc::seconds(t.utc_seconds) - m}; }
+        friend constexpr microseconds operator - ( const time_point_sec& t, const time_point_sec& m ) { return fc::seconds(t.utc_seconds) - fc::seconds(m.utc_seconds); }
+        friend constexpr microseconds operator - ( const time_point& t, const time_point_sec& m ) { return t.time_since_epoch() - fc::seconds(m.sec_since_epoch()); }
```

Comment:
I kind of like this better. Again, personal preference.

```suggestion
        constexpr microseconds time_since_epoch()const { return fc::seconds(t.utc_seconds); }
        friend constexpr time_point   operator + ( const time_point_sec& t, const microseconds& m )   { return time_point{t.time_since_epoch() + m}; }
        friend constexpr time_point   operator - ( const time_point_sec& t, const microseconds& m )   { return time_point{t.time_since_epoch() - m}; }
        friend constexpr microseconds operator - ( const time_point_sec& t, const time_point_sec& m ) { return t.time_since_epoch() - m.time_since_epoch(); }
        friend constexpr microseconds operator - ( const time_point& t, const time_point_sec& m )     { return t.time_since_epoch() - m.time_since_epoch(); }
```

Reply:
Removed all these arithmetic operators as the results could be unexpected. Better to be explicit in my opinion.